src/scope.js

```diff
@@ -400,19 +422,19 @@ class Scope {
     }
 
     __isClosed() {
-        return this.__left === null;
+        return this.__left === null || this.__left === undefined;
```

Comment:
Yes, I saw, and so you did not change that meaning of the code in those cases. Here you did, that's why I am pointing it out